Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -180,7 +180,16 @@ namespace Babylon::Polyfills::Internal
     {
         try
         {
-            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+            auto inputURL{info[1].As<Napi::String>()};
+            auto decodedURL{info.Env().Global().Get("decodeURI").As<Napi::Function>().Call({inputURL}).As<Napi::String>()};
+            auto encodedURL{inputURL};
+            // We need to percent encode the URL
+            // If decoding the URL changes something, it's already encoded - don't change anything
+            if (inputURL.StrictEquals(decodedURL))
+            {
+                encodedURL = info.Env().Global().Get("encodeURI").As<Napi::Function>().Call({inputURL}).As<Napi::String>();
+            }
+            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), encodedURL.Utf8Value());
```

Comment:
Maybe store `encodedURL` as a `Napi::Value` so that you can factor out the `.As<Napi::String>` also.